python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py

```diff
@@ -398,9 +397,14 @@ def execute_k8s_job(
             start_time=start_time,
             num_pods_to_wait_for=num_pods_to_wait_for,
         )
-    except (DagsterExecutionInterruptedError, Exception) as e:
+    except (BaseException, Exception) as e:
+        pods = api_client.get_pod_names_in_job(job_name=job_name, namespace=namespace)
+        for pod in pods:
+            pod_debug_info = api_client.get_pod_debug_info(pod_name=pod, namespace=namespace)
+            context.log.error(f"Pod debug info for pod {pod}: {pod_debug_info}")
```

Comment:
the debug info string includes the pod name. Can we use the same logic as here where we concatenate the debug info into a single log message? https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-k8s/dagster_k8s/launcher.py#L379-L390 (I think you could even consider pulling out the logic in that message into a get_job_failure_debug_info, since that method does basically exactly what you are doing here after deriving a k8s job name from the run)

(That code also includes a try catch around the get_pod_debug_info call, which i think is good to include here too - so that if there is a failure fetching the pod debug info, e.g. due to IAM permissions missing or something, it doesn't replace the original exception)